packages/next/src/cli/next-test.ts

```diff
@@ -0,0 +1,195 @@
+  }
+
+  // set the test runner. priority is CLI option > next config > default 'playwright'
+  const configuredTestRunner =
+    options?.testRunner ?? // --test-runner='foo'
+    nextConfig.experimental.defaultTestRunner ?? // { experimental: { defaultTestRunner: 'foo' }}
+    'playwright'
+
+  if (!nextConfig.experimental.testProxy) {
+    return printAndExit(
+      `\`next experimental-test\` requires the \`experimental.testProxy: true\` configuration option.`
+    )
+  }
+
+  // execute test runner specific function
+  switch (configuredTestRunner) {
+    case 'playwright':
+      return runPlaywright(baseDir, nextConfig, testRunnerArgs)
+    default:
+      return printAndExit(
+        `Test runner ${configuredTestRunner} is not supported.`
+      )
+  }
+}
+
+async function checkRequiredDeps(
+  baseDir: string,
+  testRunner: SupportedTestRunners
+) {
+  const deps = await hasNecessaryDependencies(
+    baseDir,
+    requiredPackagesByTestRunner[testRunner]
+  )
+  if (deps.missing.length > 0) {
+    await installDependencies(baseDir, deps.missing, true)
+
+    const playwright = spawn(
+      path.join(baseDir, 'node_modules', '.bin', 'playwright'),
+      ['install'],
+      {
+        cwd: baseDir,
+        shell: false,
+        stdio: 'inherit',
+        env: {
+          ...process.env,
+        },
+      }
+    )
+
+    return new Promise((resolve, reject) => {
+      playwright.on('close', (c) => resolve(c))
+      playwright.on('error', (err) => reject(err))
+    })
+  }
+}
+
+async function runPlaywright(
+  baseDir: string,
+  nextConfig: NextConfigComplete,
+  testRunnerArgs: string[]
+) {
+  await checkRequiredDeps(baseDir, 'playwright')
+
+  const playwrightConfigFile = await findUp(
+    ['playwright.config.js', 'playwright.config.ts'],
+    {
+      cwd: baseDir,
+    }
+  )
+
+  if (!playwrightConfigFile) {
+    const { pagesDir, appDir } = findPagesDir(baseDir)
+
+    const { version: typeScriptVersion } = await verifyTypeScriptSetup({
+      dir: baseDir,
+      distDir: nextConfig.distDir,
+      intentDirs: [pagesDir, appDir].filter(Boolean) as string[],
+      typeCheckPreflight: false,
+      tsconfigPath: nextConfig.typescript.tsconfigPath,
+      disableStaticImages: nextConfig.images.disableStaticImages,
+      hasAppDir: !!appDir,
+      hasPagesDir: !!pagesDir,
+    })
+
+    const isUsingTypeScript = !!typeScriptVersion
+
+    const playwrightConfigFilename = isUsingTypeScript
+      ? 'playwright.config.ts'
+      : 'playwright.config.js'
+
+    writeFileSync(
+      path.join(baseDir, playwrightConfigFilename),
+      defaultPlaywrightConfig(isUsingTypeScript)
+    )
+
+    return printAndExit(
+      `Successfully generated ${playwrightConfigFilename}. Create your first test and then run \`next experimental-test\`.`,
+      0
+    )
+  } else {
+    const playwright = spawn(
+      path.join(baseDir, 'node_modules', '.bin', 'playwright'),
```

Comment:
This doesn't work in monorepos where to `node_modules` directory lives at top level.